### DISCUSSION TITLE
Pass loader data through layout routes

### DISCUSSION THREAD

**Discussion by QuentinWidlocher** [remix-run/remix#1013] [Q&A] [answered]
Hi, is it possible to pass loader data through layout routes ?
Imagine I have a routing like that :

```
app/
├── routes/
│   ├── blog/
│   │   ├── $postId.tsx
│   │   ├── categories.tsx
│   │   ├── index.tsx
│   └── blog.tsx
│   └── index.tsx
└── root.tsx
```

Here, let's say I fetch data in `blog.tsx` that I also use in `$postId.tsx`, do I need to fetch it twice or is there a way to cleanly pass the data, like a props or something ?

**Comment by sergiodxa** [comment, marked as answer]
It’s not possible, all loader run in parallel so it would not be posible to share data between each loader, also when doing client side navigate a child loader may be called without calling the parent loaders, and if they run at the same time they could even run in different servers (like when using Serverless or CF Workers)

**Reply by sergiodxa** [reply]
One thing you could do tho is to use the useMatches hook to access the parent data inside the React component of a child route, instead of fetching in the loader.

**Reply by QuentinWidlocher** [reply]
Alright, thanks for your clear answer, it makes sense that we can't
